plugins/ExtendedTools/objmgr.c

```diff
@@ -107,22 +128,21 @@ typedef struct _DIRECTORY_ENUM_CONTEXT
 typedef struct _HANDLE_OPEN_CONTEXT
 {
     PPH_STRING CurrentPath;
-    POBJECT_ENTRY Object;
-} HANDLE_OPEN_CONTEXT, * PHANDLE_OPEN_CONTEXT;
+    PET_OBJECT_ENTRY Object;
+    PPH_STRING FullName;
+} HANDLE_OPEN_CONTEXT, *PHANDLE_OPEN_CONTEXT;
 
 typedef struct _RESOLVER_THREAD_CONTEXT
 {
-    BOOL Break;
-    POBJECT_CONTEXT Context;
-} RESOLVER_THREAD_CONTEXT, * PRESOLVER_THREAD_CONTEXT;
+    PET_OBJECT_CONTEXT Context;
+    volatile BOOLEAN Break;
+} RESOLVER_THREAD_CONTEXT, *PRESOLVER_THREAD_CONTEXT;
 
 typedef struct _RESOLVER_WORK_THREAD_CONTEXT
 {
-    POBJECT_CONTEXT Context;
-    POBJECT_ENTRY entry;
-    INT ItemIndex;
-    //BOOLEAN SortItems;
-} RESOLVER_WORK_THREAD_CONTEXT, * PRESOLVER_WORK_THREAD_CONTEXT;
+    PET_OBJECT_ENTRY entry;
+    PPH_STRING CurrentPath;
+} RESOLVER_WORK_THREAD_CONTEXT, *PRESOLVER_WORK_THREAD_CONTEXT;
```

Comment:
You could probably find an implementation that eliminates these entirely since you have access to the `PET_OBJECT_CONTEXT` through the `PET_OBJECT_ENTRY` now. If you don't do this here I'll probably try to optimize it after the merge. Eliminating these will eliminate allocations when resolving the targets.

Reply:
Implemented. Add BaseDirectory to ET_OBJECT_ENTRY.
`entry->BaseDirectory = PhReferenceObject(Context->CurrentPath);`
All needed resolver values now store inside ET_OBJECT_ENTRY and will dereference in object deletion procedure.